### ISSUE TITLE
[REQUEST] Configurable minimum upload/download graph scale

### ISSUE THREAD

**Issue by drazil100** [aristocratos/bpytop#29]
As cool as it is to have more graph showing I think the network section would be far more useful if you had the ability to set a minimum scale for the graphs to fixed values based on what you know your hardware/network is capable of. Anything that goes over the user defined minimums can scale the graph to accommodate the way it functions currently. Adding this feature would make it so your network usage doesn't look like its going through the roof when its only maxing out the 10k range.

**Comment by drazil100** [comment]
Also it would be nice to be able to set download and upload independently since usually you can download faster than you can upload.

**Comment by aristocratos** [comment]
Sure it's doable, but note that the graphs will be completely empty if current speed is less than around 5-10% percent of your set max speed depending on the height of the graph.

So if you for example have a 12.5MiB/s (100Mibps) connection and set the graph to be fixed at that value, any activity below 10Mibps would likely not show up at all.

**Comment by drazil100** [comment]
I see your point. But all the other graphs are out of 100% while the network graph is out of however much will cause the graph to take up the whole height of the graph. Either way it would be up to end end user on which option is more useful to them.

**Comment by aristocratos** [comment]
The way they work: 

There is two counters, Raise and Lower. 

If current network speed is greater than current max graph value: +1 on Raise, -1 on Lower (if Lower > 0). 

If current network speed is less than current max graph value / 10: +1 on Lower, -1 on Raise (if Raise > 0).

If Raise == 5: Set max graph value to the highest value of the last 10 values (+ 20%)

If Lower == 5: Set max graph value to the highest value of the last 10 values (* 3)

The use case for having them scale in this way is more for network activity monitoring then actual speeds.

Using them for bandwidth usage monitoring with a more fixed minimum value is definitively possible but loses a lot of the granularity because of the relatively low vertical resolution. There are only 4 possible values for each line.

**Comment by drazil100** [comment]
Its also possible you could make it some sort of "custom range" mode or something where if you have it configured you can quickly switch between granular and and not granular. 

Another option is you could have a + and - for the range where you can manually set it like you set the update ms.

I am fine though with just setting a custom range in the options like I originally suggested. These are just some alternate ideas.

**Comment by drazil100** [comment]
I just thought of another potentially interesting way of handling networking. What if the color represented how close to your top upload and download you are? The graph can keep its old functionality by the colors would scale with whatever your top recorded upload and download with your top being white as it is now. 

This way you would still be able to see the small detail of your network traffic but would be able to tell by color exactly how much of your total bandwidth you are actually using.

**Comment by aristocratos** [comment]
@drazil100 
Added in v1.0.8

**Comment by aristocratos** [comment]
> What if the color represented how close to your top upload and download you are? The graph can keep its old functionality but the colors would scale with whatever your top recorded upload and download with your top being white as it is now.

Not possible with how the graphs are created right now and to make it work there would have to be a lot of added complexity for the graph creation which most likely would have a negative effect on the cpu usage.

**Comment by drazil100** [comment]
I am actually attempting to play around with it myself which sucks cause I don't know python very well (have a lot of C# experience though) 

If I get it working and it isn't too  bad on performance ill send pr your way to take a look at. Again though I don't exactly know python so it will probably be kinda ugly xD

**Comment by drazil100** [comment]
Actually taking my head out of the code for a bit and checking out the new feature this is actually really nice being able to quick toggle the scale. Thanks!

**Comment by drazil100** [comment]
Btw I entered my top Mibps instead of my top MiB/s cause the top value is listed in Mibps and the scale was way off (obviously). Any possible way to make this less confusing to configure? I only just now realized what was happening xD

**Comment by drazil100** [comment]
![Screenshot from 2020-08-09 20-23-34](https://user-images.githubusercontent.com/17794391/89746221-32b9e700-da7e-11ea-9c6e-88d6eb69b9b6.png)
Here is a screenshot about what im talking about cause all these different units are confusing me to heck...

Using the edit I submitted PR of that scales the gradient with the top upload/download speeds it shows its clearly topping out however I set in megabytes or mebibytes (again super confused) when it displays your top speeds in the NetBox as Mebibit

This is how I configured it before and when I was confused.
![Screenshot from 2020-08-09 20-32-46](https://user-images.githubusercontent.com/17794391/89746498-8da00e00-da7f-11ea-8ed1-cffecfdd6557.png)


Anyway long story short I was confused but now im not but this could maybe be clearer to the end user by being in the same unit  as it actually displays as.

**Comment by aristocratos** [comment]
To be fair i specified in the description for the option that the value entered should be in bytes, all speeds shown in bits is in parentheses to minimize confusion. I could add another modifier for the unit entered, so you could enter "100Mbit" or "12.5Mbyte" and get the same result, and no modifier would default to bytes.

The reasoning for showing the top speeds in the box in bits instead of bytes is that ISP's almost always use bits, so it should be easier to see if you're actually reaching your ISP's claimed top speeds, and I couldn't fit both values without clipping.

I  see now what you meant with the PR you submitted, I thought you meant that there should be a separate color for every character, which would have been a pain to get working without a lot regex for appending and removing values.

**Comment by drazil100** [comment]
Closing since the feature was added and futher discussion on the scaling mechanism is happening over at #50
